src/d3a/models/myco_matcher/myco_external_matcher.py

```diff
@@ -175,6 +176,8 @@ def _get_validated_recommendations(
                     f"Market with id {recommendation.get('market_id')} doesn't exist.")
             if market.readonly:
                 # The market is already finished
+                # TODO: we're clearing markets cache after each market cycle so is this check
+                # TODO: .. really relevant? (All finished markets should be deleted already)
```

Comment:
IMO you could safely `pass` for the `market.readonly` case. Or is it important for the client to receive an exception in this case?

Reply:
It's a requirement , if any recommendation is to be placed in a finished market, we cancel the whole matching 
So either match all or don't match at all